### PR TITLE
Disable form elements while database operation is pending

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Fix issue brought up on Discourse: https://discourse.pi-hole.net/t/deleting-items-from-list-using-web-interface/27252

The issue is that quick deletion requests one after another do not always work as expected.

**How does this PR accomplish the above?:**

Disable form elements while database operation is pending to prevent interrupting running database actions. While we're at it, I moved some common code to a common file.

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/deleting-items-from-list-using-web-interface/27252/2

**Comment by DL6ER** [comment]
@XhmikosR I could need some help. Is there any documentation for the new code linter we use?

**Comment by XhmikosR** [comment]
@DL6ER xo is using ESLint behind the scenes so you can look up the errors on the eslint site. Just make sure the branch is up to date with devel though.

BTW you can run `npm run xo:fix` to fix some of the autofixable errors (doesn't apply to this PR, though).

File: scripts/pi-hole/js/groups-adlists.js [hunk 1]
```diff
@@ -5,71 +5,14 @@
  *  This file is copyright under the latest version of the EUPL.
  *  Please see LICENSE file for your rights under this license. */
 
-/* global moment:false */
+/* global showAlert:false */
```
Comment (XhmikosR) [inline scripts/pi-hole/js/groups-adlists.js]: You should combine all these into one line and separate them by comma.

**Comment by XhmikosR** [comment]
Looking though the code I see the issue. You are not making it clear that the common functions are supposed to be reused outside of the file.

One way to solve this is to do `window.foo = function` which explicitly adds the function to the window object. It's what I did in my patch https://github.com/pi-hole/AdminLTE/commit/be5c13fc7a96643c6772c1607e01d52131312a6d#diff-90cb7b827b049713469740d3c952a1c2R25

Not the best solution if we are going to have a common utils file and thus a common Object later. But we can revisit if needed.

**Comment by XhmikosR** [comment]
But seeing this I think we definitely need a `utils` object or something. I mean, when one sees [this](https://github.com/pi-hole/AdminLTE/pull/1140/files#diff-05ff8f3617deb1cfb848d6fc92892ee3R172) it's not super clear. Also don't be afraid to use more descriptive function names :)

**Comment by XhmikosR** [comment]
If you want, I could first finish my patch and make a PR and then you can build on top of that.

**Comment by XhmikosR** [comment]
Here's a quick branch https://github.com/pi-hole/AdminLTE/compare/devel...js-tweaks

I haven't tested it, but if it works let me know so that I rebase it to clean it up and you can use that as a base. There's one more instance of `showAlert` that it's not the same as the one i moved in utils, so I don't know if that could be reused or not.

We don't need to overdo with this BTW. Just some common stuff.

**Comment by DL6ER** [comment]
> We don't need to overdo with this BTW. Just some common stuff.

Yes, I was thinking the same, just the stuff the group management as this is new stuff. No need to touch the code that is known to work for anyone out there for years.

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/request-add-comment-field-to-client-table/27495/8

**Comment by PromoFaux** [comment]
Appears to do what it says on the tin. I see the inputs disable and reenable on action.

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/request-add-comment-field-to-client-table/27495/15

**Pull request merged**